runtime/protos/runtime.proto

```diff
@@ -17,232 +17,270 @@ option go_package = "github.com/ServiceWeaver/weaver/runtime/protos";
+  // Handshake.
+  EnvelopeInfo weavelet_setup_info = 2;
 
-  // A reply to the weavelet with the latest routing info.
-  RoutingInfo routing_info = 6;
+  // Envelope initiated RPC requests.
+  GetHealthRequest get_health_request = 100;
+  GetMetricsRequest get_metrics_request = 101;
+  GetLoadRequest get_load_request = 102;
+  GetProfileRequest get_profile_request = 103;
+  UpdateRoutingInfoRequest update_routing_info_request = 104;
+  UpdateComponentsRequest update_components_request = 105;
 
-  // A reply to the weavelet with the latest list of components to start.
-  ComponentsToStart components_to_start = 7;
-
-  // A reply to a GetAddressRequest.
-  GetAddressReply get_address_reply = 8;
-
-  // A reply to an ExportListenerRequest.
-  ExportListenerReply export_listener_reply = 9;
-
-  // A reply to the weavelet with an error status.
-  string error = 10;
+  // Weavelet initiated RPC replies.
+  EnsureComponentReply ensure_component_reply = 200;
+  GetAddressReply get_address_reply = 201;
+  ExportListenerReply export_listener_reply = 202;
 }
 
 // WeaveletMsg holds a message sent by a weavelet to the envelope that manages
 // it.
 message WeaveletMsg {
-  // Unique id identifies the request for which this message is a reply.
+  // A unique id for synchronous calls. A request with positive id x expects a
+  // reply with negative id -x. Non-synchronous calls have an id of 0.
   int64 id = 1;
 
-  // Exactly one of the following should be set.
-  // (Avoiding protobuf oneofs since they are annoying to use.)
+  // Exactly one of the following should be set. We avoid protobuf oneofs since
+  // they are annoying to use.
 
-  // Updated metrics.
-  MetricUpdate metrics = 2;
+  // Handshake.
+  WeaveletInfo weavelet_info = 2;
 
-  // Latest log entry.
-  LogEntry log_entry = 3;
+  // Envelope initiated RPC replies.
+  GetHealthReply get_health_reply = 100;
+  GetMetricsReply get_metrics_reply = 101;
+  GetLoadReply get_load_reply = 102;
+  GetProfileReply get_profile_reply = 103;
+  UpdateRoutingInfoReply update_routing_info_reply = 104;
+  UpdateComponentsReply update_components_reply = 105;
 
-  // Trace spans to be exported.
-  Spans trace_spans = 4;
+  // Weavelet initiated RPC requests.
+  EnsureComponentRequest ensure_component_request = 201;
+  GetAddressRequest get_address_request = 201;
+  ExportListenerRequest export_listener_request = 202;
 
-  // Component to be started.
-  ComponentToStart component_to_start = 5;
+  // Weavelet initiated messages.
+  LogEntry log_entry = 301;
+  TraceSpans trace_spans = 302;
+}
 
-  // Weavelet load to be reported.
-  WeaveletLoadReport load_report = 8;
+// EnvelopeInfo is the information provided by an envelope to a weavelet during
+// the initial envelope-weavelet handshake.
+message EnvelopeInfo {
+  string app = 1;                    // app name
+  string deployment_id = 2;          // deployment id, in UUID format
+  string id = 3;                     // weavelet id, in UUID format
```

Comment:
Can you remove this UUID requirement? The GKE deployer will plug in the pod name here, for better identifiability. Maybe just add a TODO for now?